ui/raidboss/data/02-arr/dungeon/haukke_manor.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+[{
+  zoneId: ZoneId.HaukkeManor,
+  triggers: [
+    {
+      // Dark Mist Stun with Suppression for duplicate messages
+      id: 'Haukke Normal Dark Mist Stun',
+      netRegex: NetRegexes.startsUsing({ id: '2C1', source: ['Manor Maidservant', 'Manor Claviger', 'Lady Amandine'] }),
+      condition: function(data) {
+        return data.canStun();
+      },
+      suppressSeconds: 2,
+      response: Responses.stun('info'),
+    },
+    {
+      // Soul Drain
+      id: 'Haukke Normal Steward Soul Drain Stun',
+      netRegex: NetRegexes.startsUsing({ id: '35C', source: 'Manor Steward' }),
+      condition: function(data) {
+        return data.canStun();
+      },
+      response: Responses.stun('info'),
+    },
+    {
+      // Amandine Specific Dark Mist for Classes Without Stuns
+      // Particle and spell effects in the room can make this Dark Mist hard to see.
+      // Callout ensures dodge.
+      id: 'Haukke Normal Amandine Dark Mist Dodge',
+      netRegex: NetRegexes.startsUsing({ id: '2C1', source: 'Lady Amandine', capture: false }),
+      condition: function(data) {
+        return !data.canStun();
+      },
+      response: Responses.outOfMelee('alert'),
+    },
+    {
+      // Void Fire 3
+      id: 'Haukke Normal Amandine Void Fire III',
+      netRegex: NetRegexes.startsUsing({ id: '356', source: 'Lady Amandine' }),
+      condition: function(data) {
+        return data.canInterrupt();
+      },
+      response: Responses.interrupt('info'),
+    },
+    {
+      // Void Thunder 3
+      id: 'Haukke Normal Amandine Void Thunder III',
+      netRegex: NetRegexes.startsUsing({ id: '358', source: 'Manor Claviger' }),
+      condition: function(data) {
+        return data.canInterrupt();
+      },
+      response: Responses.interrupt('info'),
+    },
+    {
+      // Void Lamp Spawn
+      id: 'Haukke Normal Void Lamps',
+      netRegex: NetRegexes.message({ line: 'The void lamps have begun emitting an eerie glow.', capture: false }),
+      infoText: function(data) {
+        return {
+          en: 'Turn off Lamps',
+        };
+      },
```

Comment:
Here and for the Handmaiden trigger, it doesn't look like you're using the `infotext` function for anything. Is this a leftover from a previous version?

Reply:
I thought I had to use infoText to return the statement as the correct type (info, alert, or alarm) to make sure its not bigger then it really needs to be. I will change it to condition though. Thank you so much for catching the CanStun and CanSilence issues, those were driving me bonkers!

Reply:
Right, no, if you don't need logic in the popup text, you can just return a bare localized object, as
```
infoText: {
  en: 'Some Unchanging Popup Text',
}
```

The only time you need to return a function instead of a bare object is the situation where you're injecting something from the `data` global object, or where you're using the `matches` object from the regex. For instance, here's the logic for [`stackMarkerOn`](https://github.com/quisquous/cactbot/blob/0712f53cdde5ae5a8e49fe22cf82c2c8a9047d45/resources/responses.js#L447):

```
  lookAwayFromTarget: (sev) => {
    let obj = {};
    obj[defaultAlertText(sev)] = (data, matches) => {
      const target = getTarget(matches);
      if (target === data.me)
        return;
      const name = isPlayerId(matches.targetId) ? data.ShortName(target) : target;
      return {
        en: 'Look Away from ' + name,
        de: 'Schau weg von ' + name,
        fr: 'Ne regardez pas ' + name,
        ja: name + 'を見ない',
        cn: '背对' + name,
        ko: name + '에게서 뒤돌기',
      };
    };
    return obj;
  },
```

As you can see, the function takes both `data` and `matches` as arguments, processes `matches` to find the name of the target, and then uses the `ShortName` function from `data` to return a localized object with a customized target name.

Sometimes you'll have a parameter stored on the `data` object but not need the `match` parameter. Let's say that the boss can cast Half Vanish, which makes either the north or the south side of the room unsafe. Which half depends on whether the boss has previously cast Northern Cross or Southern Cross. (This is all made-up for this example. of course.) So you would have one trigger that checks for which Cross is cast, (let's say North,) and that trigger would store it as `data.north = true;`. Then, when the boss casts Half Vanish, you would set the trigger up like this:

```
{
  id: 'Made-up Half Vanish',
  netRegex: NetRegexes.startsUsing({ id: '9876', source: 'Giant Monster', capture: false }),
  alertText: function(data) {
    if (data.north) {
      return {
        /en: 'Go North!',
      };
    }
    return {
      en: 'Go South!',
    }
  }
}
```
Note that regex capturing isn't needed, since we don't care about the particulars of the regex. We just care that there was a match at all.

Sorry for the wall of text. This is all stuff that's easy to use once you know it, but it can be tricky to step through the pieces. There are a lot of details!